pkg/csi/service/vanilla/controller.go

```diff
@@ -385,8 +384,14 @@ func (c *controller) createBlockVolume(ctx context.Context, req *csi.CreateVolum
 	}
 	volSizeMB := int64(common.RoundUpSize(volSizeBytes, common.MbInBytes))
 
-	// Check if the feature state of block-volume-snapshot is enabled
+	// Check if the feature states are enabled.
 	isBlockVolumeSnapshotEnabled := commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.BlockVolumeSnapshot)
+	isImprovedVolumeTopologyEnabled := commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx,
```

Comment:
Can we move feature-gate retrieval in the controller init

Reply:
I will take it up as an improvement in a follow-up PR